lib/order.rb

```diff
@@ -0,0 +1,96 @@
+class Order
+  
+  attr_reader :id  
+  attr_accessor :customer, :products, :fulfillment_status
+  
+  def initialize(id, products, customer, fulfillment_status = :pending)
+    @id = id
+    @products = products
+    @customer = customer
+    @fulfillment_status = fulfillment_status
+    fulfillment_statuses = [:pending, :paid, :processing, :shipped, :complete]
+    if !fulfillment_statuses.include? @fulfillment_status
+      raise ArgumentError
```

Comment:
A message saying what the bad status is would be helpful here.

```suggestion
      raise ArgumentError.new("Invalid fulfillment status: #{@fulfillment_status}")
```